Dockerfile

```diff
@@ -0,0 +1,72 @@
+# Based on OpenAI's mujoco-py Dockerfile
+
+FROM nvidia/cuda:10.0-runtime-ubuntu18.04 AS base
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN echo ttf-mscorefonts-installer msttcorefonts/accepted-mscorefonts-eula select true | debconf-set-selections \
+    && apt-get update -q \
+    && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    ffmpeg \
+    git \
+    libgl1-mesa-dev \
+    libgl1-mesa-glx \
+    libglew-dev \
+    libosmesa6-dev \
+    net-tools \
+    parallel \
+    python3.7 \
+    python3.7-dev \
+    python3-pip \
+    rsync \
+    software-properties-common \
+    unzip \
+    vim \
+    virtualenv \
+    xpra \
+    xserver-xorg-dev \
+    ttf-mscorefonts-installer \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN curl -o /usr/local/bin/patchelf https://s3-us-west-2.amazonaws.com/openai-sci-artifacts/manual-builds/patchelf_0.9_amd64.elf \
+    && chmod +x /usr/local/bin/patchelf
+
+ENV LANG C.UTF-8
+
+RUN    mkdir -p /root/.mujoco \
+    && curl -o mjpro150.zip https://www.roboti.us/download/mjpro150_linux.zip \
+    && unzip mjpro150.zip -d /root/.mujoco \
+    && rm mjpro150.zip
+
+# Set the PATH to the venv before we create the venv, so it's visible in base.
+# This is since we may create the venv outside of Docker, e.g. in CI
+# or by binding it in for local development.
+ENV PATH="/imitation/venv/bin:$PATH"
+ENV LD_LIBRARY_PATH /usr/local/nvidia/lib64:/root/.mujoco/mjpro150/bin:${LD_LIBRARY_PATH}
+
+FROM base as python-req
```

Comment:
I'm guessing that `base` is used here for caching purposes. Should `python-req` used for development, and `full` used for CI?

Not sure why `python-req` is useful to have as a separate build.

Reply:
The intention here is:
  - `base` is used in a CI build.
  - `python-req` is used for local development, where you can bind the source code using `-v` from outside Docker.
  - `full` is used if you want to deploy the finished product, e.g. to Ray or to run tests on a local machine.

The reason I'm using `base` rather than `full` in a CI build is to avoid having to rebuild the Docker image for each build. This is possible but it tends to be slow and is quite a pain to set up. Instead we rebuild the `venv` leveraging CircleCI's built-in caching support.

Reply:
(I've added comments documenting this now.)